lightswitch-object/src/object.rs

```diff
@@ -35,33 +35,30 @@ pub struct ElfLoad {
 
 #[derive(Debug)]
 pub struct ObjectFile<'a> {
-    leaked_mmap_ptr: *const Mmap,
+    /// Warning! `object` must always go above `mmap` to ensure it will be dropped
+    /// before. Rust guarantees that fields are dropped in the order they are defined.
     object: object::File<'a>,
```

Comment:
Should this lifetime not be:
```suggestion
    object: object::File<'static>,
```

With a comment explaining that this lifetime is attached to the lifetime of `self.mmap`?